packages/builder/src/components/design/PropertiesPanel/PropertyControls/ColumnEditor/ColumnEditor.svelte

```diff
@@ -18,22 +18,30 @@
   let boundValue
 
   $: datasource = getDatasourceForProvider($currentAsset, componentInstance)
-  $: schema = getSchemaForDatasource($currentAsset, datasource).schema
+  $: schema = getSchema($currentAsset, datasource)
   $: options = Object.keys(schema || {})
   $: sanitisedValue = getValidColumns(value, options)
   $: updateBoundValue(sanitisedValue)
 
+  const getSchema = (asset, datasource) => {
+    const schema = getSchemaForDatasource(asset, datasource).schema
+
+    // Don't show ID and rev in tables
+    if (schema) {
+      delete schema._id
```

Comment:
Hopefully not an issue but just thought I'd check - is there a chance here that the schema gets mutated, and re-saved to the DB? I remember something like this happened before, and it caused some issues.

Reply:
Well remembered and yea it makes sense to double check! In this case we're all good, because `getSchemaForDatasource` always ensures to deep clone any real table schemas to ensure they aren't mutated.

The resulting schema that we delete these columns from is then only ever used in these components to determine which column options to make available, but is never saved or used for anything else.

Good call on double checking though.